src/material.js

```diff
@@ -215,29 +214,45 @@ export class Material {
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
-    calculateHash(salt = '', isScaled = false) {
+    calculateHash(salt = "", isScaled = false) {
         let message;
         if (!this.isNonPeriodic) {
-            message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+            message =
+                this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
         } else {
             message = this.getInchiStringForHash();
         }
         return CryptoJS.MD5(message).toString();
     }
 
+    /**
+     * Function calculates the scaledHash for a material by calling the calculateHash function and passing it the proper
+     * salt String and isScaled Boolean value.
+     * @param {String} salt
+     * @param {Boolean} isScaled
+     * returns {String}
+     */
+    calculateScaledHash(salt = "", isScaled = true) {
+        return this.calculateHash(salt, isScaled);
+    }
+
     set hash(hash) {
-        this.setProp('hash', hash);
+        this.setProp("hash", hash);
```

Comment:
There was a reason why we had the previous logic that avoided directly storing the hash and recalculated it "on-the-fly" from lattice/basis - it was to avoid a "corrupted" state where the lattice and basis are changed but a hash is not.

That's why we didn't set the `hash` in a constructor (https://github.com/Exabyte-io/made.js/pull/53/files#diff-c664f5e378394d9c3fc8b008011f236c1beb1a8f1a2f741e86dd99712b970a32R59) and toJSON in https://github.com/Exabyte-io/made.js/pull/53/files#diff-c664f5e378394d9c3fc8b008011f236c1beb1a8f1a2f741e86dd99712b970a32R74 does not return it either. Hash is a "second-tier" variable.

When introducing inchi, we have to understand how this logic can be adjusted.

One way would be to keep the inchi value inside the basis to make sure they both are always updated together.

Otherwise, we have to understand how to assert a correct state of basis vs hash relationship at all times with some other checks.